src/api/index.ts

```diff
@@ -363,6 +363,34 @@ export const getUnclaimedSellerFunds = async (pair: TokenPair, index?: Index, ac
   }
 }
 
+export const getUnclaimedSellerFundsFromSeveral = async (
+  sell: DefaultTokenObject,
+  buy: DefaultTokenObject,
+  userAccount?: Account,
+  indices: number = 0,
+) => {
+  const { DutchX } = await promisedAPI
+  userAccount = await fillDefaultAccount(userAccount)
+  const sellArr = [], buyArr = []
+
+  const claimableIndices = (await DutchX.getIndicesWithClaimableTokensForSellers(sell.address, buy.address, userAccount, indices))[0].map((i: BigNumber) => i.toNumber())
+  const claimableIndicesBalance = (await DutchX.getIndicesWithClaimableTokensForSellers(sell.address, buy.address, userAccount, indices))[1].map((i: BigNumber) => i.toNumber())
+
+  const lastIndexCleared = ((await DutchX.getClosingPrice({ sell, buy }, claimableIndices[claimableIndices.length - 1])).map(i => i.toNumber()))[0] > 0
+
+  console.log('claimableIndices => ', claimableIndices, 'claimableIndicesBalance => ', claimableIndicesBalance, 'lastIndexCleared? => ', lastIndexCleared)
+
+  for (let i = 1; i <= claimableIndices.length; ++i) {
+    if (!lastIndexCleared) continue
```

Comment:
You want to include `sell/buy.address` for all indices except for the last if it happens to be the ongoing auction not yet closed, right? `if (!lastIndexCleared) continue` will end up in empty arrays should it trigger.

What about:
```
const sellArr = Array.from({length: lastIndexCleared ? claimableIndices.length : claimableIndices.length - 1}, () => sell.address)
const buyArr = ...
```
This will give for
```
claimableIndices [1,3,5] and lastIndexCleared = true  =>  sellArr = [address, address, address]

claimableIndices [1,3,5] and lastIndexCleared = false =>  sellArr = [address, address]
```

Reply:
cool, nice optimization. will change